llvm/lib/Transforms/Utils/CloneFunction.cpp

```diff
@@ -322,16 +317,21 @@ void llvm::CloneFunctionInto(Function *NewFunc, const Function *OldFunc,
   DISubprogram *SPClonedWithinModule =
       CollectDebugInfoForCloning(*OldFunc, Changes, DIFinder);
 
-  ModuleLevelChanges =
-      BuildDebugInfoMDMap(VMap.MD(), Changes, DIFinder, SPClonedWithinModule);
+  MetadataSetTy IdentityMD;
+  FindDebugInfoToIdentityMap(IdentityMD, Changes, DIFinder,
+                             SPClonedWithinModule);
 
-  const auto RemapFlag = ModuleLevelChanges ? RF_None : RF_NoModuleLevelChanges;
+  // Current implementation always upgrades from local changes to module level
+  // changes due to the way metadata cloning is done. See
+  // BuildDebugInfoToIdentityMap for more details.
```

Comment:
I can't find `BuildDebugInfoToIdentityMap` in this patch

Reply:
Also note how this comment reads: it talks about "upgrading local changes to module level changes", but there is nothing about the upgrade in the final code (as opposed to the  old code).  In other words, this comment is making a temporal reference to code that used to exist, which is not very helpful for readers.

We can phrase it as: "Cloning is always a Module level operation, since Metadata needs to be cloned".